include/sycl/algorithm/inner_product.hpp

```diff
@@ -34,7 +34,9 @@
 #include <iostream>
 
 #include <sycl/algorithm/algorithm_composite_patterns.hpp>
+#include <sycl/algorithm/reduce.hpp>
 #include <sycl/helpers/sycl_differences.hpp>
+#include <boost/iterator/zip_iterator.hpp>
```

Comment:
This causes Parallel STL to require boost when building, which was not there before. Can you make this include only for triSYCL (since I guess triSYCL uses boost so there is no difference)?